### DISCUSSION TITLE
AdvancedProfanityFilter does not detect words of different fonts I think?

### DISCUSSION THREAD

**Discussion by MaryAtlas** [FrostCo/AdvancedProfanityFilter#582] [Q&A] [answered]
Hello. I am using Chrome browser with the extension installed, and with all modes tried, Normal, Advanced, Deep, it would not filter the bolded words. Here is the exact link where it occurred. 
https://shopee.com.my/product/74423802/25055503788
If you scroll down to the Product Description paragraph, the bolded words are 
"𝐁𝐞𝐬𝐭 𝐓𝐫𝐚𝐯𝐞𝐥 𝐏𝐚𝐫𝐭𝐧𝐞𝐫 𝐈 𝐒𝐚𝐟𝐞 & 𝐄𝐚𝐬𝐲 𝐔𝐬𝐞
𝐋𝐞𝐭'𝐬 𝐏𝐥𝐚𝐧 𝐘𝐨𝐮𝐫 𝐇𝐨𝐥𝐢𝐝𝐚𝐲 𝐰𝐢𝐭𝐡 𝐓𝐫𝐚𝐯𝐞𝐥𝐥𝐨𝐫 𝐍𝐨𝐰".
Not a single word in these sentences could be filtered by AdvancedProfanityFilter.
What is going on?

P.S: Try opening the link in incognito mode if it redirects you to the sign up page.

**Comment by richardfrost** [comment]
@MaryAtlas, The issue is that those characters are actually not "normal" letters. They look like it, but they are actually special Unicode characters. I'll have to see if I can find a good way to try and match characters that are close/look like other characters.

I found a link to avoid the sign-up page [here](https://shopee.com.my/Hidden-Camera-Detector-With-Motion-Sensor-Alarm-Pengesan-Camera-Tersembunyi-%E9%98%B2%E5%81%B7%E6%8B%8D%E6%8E%A2%E6%B5%8B%E4%BB%AA-i.74423802.25055503788).

**Reply by MaryAtlas** [reply]
Hmm? Do you mean that AdvancedProfanityFilter could not filter special Unicode characters? It does not support them?
I actually copied the exact special Unicode words from the text and added it into filter. And it did not work. 
![Capture](https://github.com/user-attachments/assets/47027f6e-b7da-4d1b-a488-20401dac1587)

I don't mind if actual words like B did not filter special "𝐁". But shouldn't it filter "𝐁" if I add exactly "𝐁" into filter?

Awesome, can't wait to continue using the awesome tool that you created.

**Reply by richardfrost** [reply]
@MaryAtlas, If you copy-paste the characters, it should be able to filter them correctly. For example, I added the word `𝐁𝐞𝐬𝐭` to the filter and it filtered the text. Is it not working that way for you? If it isn't filtering on the Shopee page, does it filter here on this Github page (since it was added in the discussion)? Also, if it isn't working, could you share what the "Word" options are for it, like this:

![image](https://github.com/user-attachments/assets/ab2f2a6f-fb07-4994-999d-5105638095bb)

**Reply by MaryAtlas** [reply]
Yes, it is not working like yours. Here are the screenshots.
![1](https://github.com/user-attachments/assets/67176d53-474e-4f03-9ace-b82c1c66fe1d)
![2](https://github.com/user-attachments/assets/682193e8-065e-4db9-86ad-4bc7dbc8fb61)
I tried adding 𝐁𝐞𝐬𝐭, then I also tried adding 𝐓𝐫𝐚𝐯𝐞𝐥. I tried all 3 modes Normal, Advanced, Deep, nothing works. It still shows.


While it filtered the non-special words as usual, like in the screenshot where I filtered "be" in "could be filtered".

**Reply by MaryAtlas** [reply]
While waiting for you, I tried "RESTORE DEFAULTS", then solely adding 𝐓𝐫𝐚𝐯𝐞𝐥. It worked and filtered it. However every time I import my backup.json, it goes back to not working. The difference I noticed between them is on a fresh default, when I add 𝐓𝐫𝐚𝐯𝐞𝐥, it is displayed in *** in Words & Phrases dropdown. On my backup.json, it displays as plain 𝐓𝐫𝐚𝐯𝐞𝐥 in Words & Phrases dropdown. Please help for me.

**Comment by MaryAtlas** [comment]
So this is obviously a bug in your program. Do you want to open a bug report? Are you able to fix it?

**Reply by richardfrost** [reply]
@MaryAtlas, We can definitely create an issue for this bug, but I will probably need to see a copy of your `.json` config file to figure out what settings are causing the issue. Can you share a copy (either here, or you can email it to me).

**Reply by MaryAtlas** [reply]
I have emailed the .json file to richard@richardfrost.info
Please take a look.

**Reply by richardfrost** [reply, marked as answer]
Thanks for sending that over @MaryAtlas. I think it helped me to figure out where the issue is, but it would be great if you could confirm it. Can you see if it works if you uncheck these options for the "𝐁𝐞𝐬𝐭" word? I think there is a conflict there with these special Unicode characters and those options that I'll have to look into more.

![image](https://github.com/user-attachments/assets/f8e45bb1-f002-41ee-a245-2a91894e4396)

**Reply by MaryAtlas** [reply]
Oh my, yes it does fix the issue! It is finally filtered. You are a genius. Thank you for the hard work!

**Reply by richardfrost** [reply]
@MaryAtlas I'm so glad that it is working again for you. I have created an [issue](https://github.com/FrostCo/AdvancedProfanityFilter/issues/591) with these details. I don't have a lot of time to work on it right now, but at least its on the list now, and I'll work on it when I can. Thanks for you help and patience!
